src/main/surfrdMod.F90

```diff
@@ -882,6 +887,130 @@ subroutine surfrd_veg_dgvm(begg, endg)
+         dim1name=grlnd, readvar=readvar)
+    if (.not. readvar) then
+       call endrun( msg=' ERROR: nhillcolumns not on surface data file'//errMsg(sourcefile, __LINE__))
+    else
+       ncolumns_hillslope(begg:endg) = arrayl(begg:endg)
+    endif
+    deallocate(arrayl)
+
+    ! pft_from_file and pft_lowland_upland assume that 1 pft
+    ! will exist on each hillslope column.  In prepration, set one
+    ! pft weight to 100 and the rest to 0.  The vegetation type
+    ! (patch%itype) will be reassigned when initHillslope is called later.
+    if(pft_distribution_method == pft_from_file .or. &
+         pft_distribution_method == pft_lowland_upland) then
+       do g = begg, endg
+          ! If hillslopes will be used in a gridcell, modify wt_nat_patch, otherwise use original patch distribution
+          if(ncolumns_hillslope(g) > 0) then
+             ! First patch gets 100% weight; all other natural patches are zeroed out
+             wt_nat_patch(g,:)         = 0._r8
+             wt_nat_patch(g,natpft_lb) = 100._r8
+          endif
+       enddo
+    endif
+
+    ! pft_uniform_dominant_pft uses the patch with the
+    ! largest weight for all hillslope columns in the gridcell
+    if (pft_distribution_method == pft_uniform_dominant_pft) then
+       allocate(max_indices(1))
+       do g = begg, endg
+          ! If hillslopes will be used in a gridcell, modify wt_nat_patch,
+          ! otherwise use original patch distribution
+          if(ncolumns_hillslope(g) > 0) then
+
+             call find_k_max_indices(wt_nat_patch(g,:),natpft_lb,1,max_indices)
+             wt_nat_patch(g,:) = 0._r8
+             wt_nat_patch(g,max_indices(1)) = 100._r8
+             
+          endif
+       enddo
+       deallocate(max_indices)
+    endif
+
+    ! pft_lowland_dominant_pft uses the two patches with the
+    ! largest weights for the hillslope columns in the gridcell
+    if (pft_distribution_method == pft_lowland_dominant_pft) then
+       allocate(max_indices(2))
+       do g = begg, endg
+          ! If hillslopes will be used in a gridcell, modify wt_nat_patch, otherwise use original patch distribution
+          if(ncolumns_hillslope(g) > 0) then
+             
+             ! Preserve the relative weights of the largest and
+             ! next largest weights using arbitrarily chosen values
+             ! (i.e. 1 should be larger than 2) that sum to 100.
+             ! This will minimize memory usage while still allowing
+             ! HillslopeDominantLowlandPft to pick out the two largest patch types.
+
+             call find_k_max_indices(wt_nat_patch(g,:),natpft_lb,2,max_indices)
+             ! check that 2nd index weight is non-zero
+             if (wt_nat_patch(g,max_indices(2)) > 0._r8) then
+                wt_nat_patch(g,:) = 0._r8
+                wt_nat_patch(g,max_indices(1)) = 75._r8
+                wt_nat_patch(g,max_indices(2)) = 25._r8
```

Comment:
Why 75 and 25 instead of preserving their weights relative to one another?

Reply:
Per @swensosc:
> this just took less code and didn't require any additional variables or calculations to create two weight that summed to 100.  It could be changed if it seemed more clear to do so.
I'll plan to do this.